src/stream/src/executor/row_id_gen.rs

```diff
@@ -93,16 +101,26 @@ impl RowIdGenExecutor {
         #[for_await]
         for msg in upstream {
             let msg = msg?;
-            if let Message::Chunk(chunk) = msg {
-                // For chunk message, we fill the row id column and then yield it.
-                let (ops, mut columns, bitmap) = chunk.into_inner();
-                columns[self.row_id_index] = self
-                    .gen_row_id_column_by_op(&columns[self.row_id_index], &ops)
-                    .await;
-                yield Message::Chunk(StreamChunk::new(ops, columns, bitmap));
-            } else {
-                // For barrier message or watermark message, we just yield it.
-                yield msg;
+
+            match msg {
+                Message::Chunk(chunk) => {
+                    // For chunk message, we fill the row id column and then yield it.
+                    let (ops, mut columns, bitmap) = chunk.into_inner();
+                    columns[self.row_id_index] = self
+                        .gen_row_id_column_by_op(&columns[self.row_id_index], &ops)
+                        .await;
+                    yield Message::Chunk(StreamChunk::new(ops, columns, bitmap));
+                }
+                Message::Barrier(barrier) => {
+                    // Update row id generator if vnode mapping is changed.
+                    // Note that: since update barrier will only occurs between pause and resume
+                    // barrier, duplicated row id won't be generated.
```

Comment:
```suggestion
                    // Update row id generator if vnode mapping changes.
                    // Note that: since update barrier only exists between pause and resume
                    // barrier, duplicated row id won't be generated.
```